evalml/pipelines/components/transformers/encoders/target_encoder.py

```diff
@@ -0,0 +1,67 @@
+from ..transformer import Transformer
+
+from evalml.pipelines.components import ComponentBaseMeta
+from evalml.utils import import_or_raise
+
+
+class TargetEncoderMeta(ComponentBaseMeta):
+    """A version of the ComponentBaseMeta class which includes validation on an additional target-encoder method `categories`"""
+    METHODS_TO_CHECK = ComponentBaseMeta.METHODS_TO_CHECK + ['get_feature_names']
+
+
+class TargetEncoder(Transformer, metaclass=TargetEncoderMeta):
+    """Target encoder to encode categorical data"""
+    name = 'Target Encoder'
+    hyperparameter_ranges = {}
+
+    def __init__(self,
```

Comment:
Should we file an issue for adding a `top_n` like the OneHotEncoder? I like @rpeck 's suggestion to bin low-frequency categories into an "other" category.

Reply:
I'm not sure if we want to do `top_n` for this. I remember in our discussion we talked about using `OHE` for smaller amounts of categorical values (for instance, < 100), while using `TE` for larger amounts of categorical values. I think this was something we would look into further, and I can file an issue for that exploration? Correct me if I'm wrong @dsherry @rpeck

Reply:
Yea I don't think we should drop categories based on top_n like the OHE, rather that we group all the ones outside the top_n most frequent into the same category. 

Definitely something for future work! Just wanted to make sure we tracked it.

Reply:
Submitted an issue [here](https://github.com/alteryx/evalml/issues/1440)!